From: Julia Lawall <julia@diku.dk>
Subject: [PATCH 062/111]     net/9p/trans_fd.c: Fix unsigned return type
Git-commit: 29af9309dba25077f711f5f975977714bc43a0c3
Patch-Mainline: 2.6.36
References: FATE#311639

    The function has an unsigned return type, but returns a negative
constant
    to indicate an error condition.  The result of calling the function is
    always stored in a variable of type (signed) int, and thus unsigned can
be
    dropped from the return type.

    A sematic match that finds this problem is as follows:
    (http://coccinelle.lip6.fr/)

    // <smpl>
    @exists@
    identifier f;
    constant C;
    @@

     unsigned f(...)
     { <+...
    *  return -C;
     ...+> }
    // </smpl>

	upstream commit 29af9309dba25077f711f5f975977714bc43a0c3

    Signed-off-by: Julia Lawall <julia@diku.dk>
    Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Harsh Prateek Bora <harsh@linux.vnet.ibm.com>
Acked-by: Bruce Rogers <brogers@suse.com>
---
 net/9p/trans_fd.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/net/9p/trans_fd.c b/net/9p/trans_fd.c
index 38e6c64..35b9198 100644
--- a/net/9p/trans_fd.c
+++ b/net/9p/trans_fd.c
@@ -221,7 +221,7 @@ static void p9_conn_cancel(struct p9_conn *m, int err)
 	}
 }
 
-static unsigned int
+static int
 p9_fd_poll(struct p9_client *client, struct poll_table_struct *pt)
 {
 	int ret, n;
-- 
1.7.1.1

